simphony_osp/ontology/individual.py

```diff
@@ -2067,19 +2033,23 @@ def attributes_add(
         # TODO: prevent the end result having more than one value depending on
         #  ontology cardinality restrictions and/or functional property
         #  criteria.
-        for value in filter(lambda v: v is not None, values):
+        values = set(
+            attribute.convert_to_datatype(value)
+            for value in values
+            if value is not None
+        )
+        for value in values:
```

Comment:
Won't this iterate the values twice? (Not sure if it might be the same generator issue as last time)

Reply:
Well spotted, thanks. Just sent a commit fixing it. If the user provides repeated attributes it is should not be a problem, as the graph is a _set_ (from set theory) of triples.